io-engine/src/lvs/lvs_store.rs

```diff
@@ -939,6 +956,31 @@ impl Lvs {
         Ok(())
     }
 
+    /// When the underlying AIO bdev is resized, crypto bdev receives SPDK_BDEV_EVENT_RESIZE
+    /// event. The crypto bdev then adjusts its block count accordingly. To ensure that this resize
+    /// operation completes before proceeding, we wait briefly for the crypto bdev to update.
+    /// This delay gives the crypto bdev time to process the resize event asynchronously.
+    async fn crypto_vbdev_resized(&self) -> bool {
+        for _i in 1..=6 {
+            let base_bdev = self.base_bdev();
+            let disk_bdev = base_bdev
+                .crypto_base_bdev()
+                .map(Bdev::new)
+                .unwrap_or_else(|| self.base_bdev());
+            let disk_bdev_size = disk_bdev.num_blocks() * disk_bdev.block_len() as u64;
+            let crypto_bdev_size = base_bdev.num_blocks() * base_bdev.block_len() as u64;
+            if crypto_bdev_size == disk_bdev_size {
+                return true;
+            } else {
+                let rx = mayastor_sleep(std::time::Duration::from_millis(500));
```

Comment:
How long does it typically take? If less then it might be worth sleeping less time with more max retries to avoid blocking `gRPC`

```suggestion
                let rx = mayastor_sleep(std::time::Duration::from_millis(100));
```

Reply:
It completes at 2nd iteration. So takes more then 500ms.

Reply:
I guess you don't know if it completes at 2nd iteration? You slept 500ms but it could have completed anytime?

Reply:
No i had added loggers before just to see both bdev size. I'm okay with sleeping less time with more retries.

Reply:
Let's use less sleep time with more retries then